packages/react-button/src/components/CompoundButton/useCompoundButtonStyles.ts

```diff
@@ -0,0 +1,271 @@
+      },
+    };
+  },
+  secondaryContentLarge: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      fontSize: compoundButtonTokens.large.secondaryContentFontSize,
+    };
+  },
+  secondaryContentPrimary: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      color: compoundButtonTokens.primary.secondaryContentColor,
+
+      ':hover': {
+        color: compoundButtonTokens.primary.hovered?.secondaryContentColor,
+      },
+
+      ':active': {
+        color: compoundButtonTokens.primary.pressed?.secondaryContentColor,
+      },
+    };
+  },
+  secondaryContentDisabled: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      color: compoundButtonTokens.disabled.secondaryContentColor,
+
+      ':hover': {
+        color: compoundButtonTokens.disabled.hovered?.secondaryContentColor,
+      },
+
+      ':active': {
+        color: compoundButtonTokens.disabled.pressed?.secondaryContentColor,
+      },
+    };
+  },
+});
+
+export const useCompoundButtonStyles = (state: CompoundButtonState, selectors: CompoundButtonStyleSelectors) => {
+  // Save the class names used in useButtonStyles and undefine them at the state level so that they are always applied
+  // last.
+  const {
+    className: rootClassName,
+    children: { className: childrenClassName } = { className: undefined },
+    icon: { className: iconClassName } = { className: undefined },
+  } = state;
+  state.className = undefined;
+  if (state.children) {
+    state.children.className = undefined;
+  }
+  if (state.icon) {
+    state.icon.className = undefined;
+  }
+  useButtonStyles(state, selectors);
+
+  const styles = useStyles();
+
+  state.className = ax(
+    state.className,
+    styles.root,
+    selectors.size === 'small' && styles.rootSmall,
+    selectors.size === 'large' && styles.rootLarge,
+    selectors.iconOnly && styles.rootIconOnly,
+    selectors.iconOnly && selectors.size === 'small' && styles.rootIconOnlySmall,
+    selectors.iconOnly && selectors.size === 'large' && styles.rootIconOnlyLarge,
+    rootClassName,
+  );
+
+  if (state.children) {
+    state.children.className = ax(
+      state.children.className,
+      selectors.size === 'small' && styles.childrenSmall,
+      childrenClassName,
+    );
+  }
+
+  if (state.icon) {
+    state.icon.className = ax(state.icon.className, styles.icon, iconClassName);
+  }
+
+  if (state.contentContainer) {
+    state.contentContainer.className = ax(styles.contentContainer, state.contentContainer.className);
+  }
+
+  if (state.secondaryContent) {
+    state.secondaryContent.className = ax(
+      styles.secondaryContent,
+      selectors.size === 'large' && styles.secondaryContentLarge,
+      selectors.primary && styles.secondaryContentPrimary,
+      selectors.disabled && styles.secondaryContentDisabled,
+      state.secondaryContent.className,
```

Comment:
I'm don't know the internals of what `ax` does, but I see some inconsistencies in the ordering of the input. In the root, I see that `state.className` (the user's input) comes as the first param. Same with icon. But with `secondaryContent`, the user's input comes last.

Is that intended? Does it matter, is there an ordering here to ensure user overrides are more important?

Reply:
It is intended, it should normally come in the end but it's coming here first because that's actually the result of the `useButtonStyles` call before that. If you take a closer look I actually get the `classnames` that are applied at the end at the beginning of the function.